[cocoa] Change color profile of HTTPSOnly warning

sysrqb opened the pull request:
#### e4825f2fe0bd944db35d381c16123fe6c6bcfc8c
<pre>
[cocoa] Change color profile of HTTPSOnly warning
<a href="https://bugs.webkit.org/show_bug.cgi?id=277780">https://bugs.webkit.org/show_bug.cgi?id=277780</a>
<a href="https://rdar.apple.com/133420418">rdar://133420418</a>

Reviewed by Alex Christensen.

This patch uses gray backgrounds for the warning, instead of red, and
associated text colors.

Also use const-ref parameters in another switchOn.

* Source/WebKit/HTTPSBrowsingWarning.xcassets/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSBackground.colorset/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSBrowsingWarningText.colorset/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSBrowsingWarningTitle.colorset/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSWarningBoxBackground.colorset/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSWarningText.colorset/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSWarningTitle.colorset/Contents.json: Added.
* Source/WebKit/UIProcess/Cocoa/_WKWarningView.h:
* Source/WebKit/UIProcess/Cocoa/_WKWarningView.mm:
(colorForItem):
(-[_WKWarningView initWithFrame:browsingWarning:completionHandler:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/282025@main">https://commits.webkit.org/282025@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1f0acd4bc5e706393cd535a78b090853a9286d16

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61763 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41117 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/14355 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/65742 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/12308 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48803 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/12579 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/49811 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/8544 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/64832 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/38190 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/53497 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30643 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/34847 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/11239 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56652 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/11030 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/67470 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5706 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/10789 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/57187 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5731 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/53444 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/57420 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4694 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9311 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/36917 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/38001 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/39097 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/37746 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/UIProcess/Cocoa/_WKWarningView.mm:
```diff
@@ -119,15 +120,31 @@
-    case WarningItem::SafeBrowsingBackground:
-        return colorNamed(@"WKSafeBrowsingWarningBackground");
+    case WarningItem::BrowsingWarningBackground:
+        return WTF::switchOn(warningView.warning->data(), [&] (WebKit::BrowsingWarning::SafeBrowsingWarningData) {
```
const SafeBrowsingWarningData&

Otherwise a copy constructor is called.  Same for all.

I also don't think we need to capture anything.

Commit:
[cocoa] Change color profile of HTTPSOnly warning
https://bugs.webkit.org/show_bug.cgi?id=277780
rdar://133420418

Reviewed by Alex Christensen.

This patch uses gray backgrounds for the warning, instead of red, and
associated text colors.

Also use const-ref parameters in another switchOn.

* Source/WebKit/HTTPSBrowsingWarning.xcassets/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSBackground.colorset/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSBrowsingWarningText.colorset/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSBrowsingWarningTitle.colorset/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSWarningBoxBackground.colorset/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSWarningText.colorset/Contents.json: Added.
* Source/WebKit/HTTPSBrowsingWarning.xcassets/WKHTTPSWarningTitle.colorset/Contents.json: Added.
* Source/WebKit/UIProcess/Cocoa/_WKWarningView.h:
* Source/WebKit/UIProcess/Cocoa/_WKWarningView.mm:
(colorForItem):
(-[_WKWarningView initWithFrame:browsingWarning:completionHandler:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/282025@main
